src/commands/ImportObjectsCommandGenerator.js

```diff
@@ -29,102 +29,75 @@ const ANSWERS_NAMES = {
-			LIST_OBJECTS_COMMAND_NAME
-		);
+		this._listObjectsMetadata = commandsMetadataService.getCommandMetadataByName(LIST_OBJECTS_COMMAND_NAME);
 	}
 
-	_getCommandQuestions(prompt) {
-		const questions = this._generateListObjectQuestions();
-
-		return new Promise((resolve, reject) => {
-			prompt(questions)
-				.then(firstAnswers => {
-					const paramsForListObjects = this._arrangeAnswersForListObjects(firstAnswers);
-					const executionContextForListObjects = new SDKExecutionContext({
-						command: this._listObjectsMetadata.name,
-						params: paramsForListObjects,
-						includeProjectDefaultAuthId: true
-					});
-
-					executeWithSpinner({
-						action: this._sdkExecutor.execute(executionContextForListObjects),
-						message: TranslationService.getMessage(MESSAGES.LOADING_OBJECTS),
-					})
-						.then(operationResult => {
-							const { data } = operationResult;
-							if (SDKOperationResultUtils.hasErrors(operationResult)) {
-								SDKOperationResultUtils.logResultMessage(operationResult);
-								SDKOperationResultUtils.logErrors(operationResult);
-								return;
-							}
-							if (Array.isArray(data) && operationResult.data.length === 0) {
-								NodeUtils.println(
-									TranslationService.getMessage(MESSAGES.NO_OBJECTS_TO_LIST),
-									NodeUtils.COLORS.RESULT
-								);
-								return;
-							}
-
-							const questions = this._generateSelectionObjectQuestions(
-								operationResult
-							);
-
-							prompt(questions).then(secondAnswers => {
-								const combinedAnswers = { ...firstAnswers, ...secondAnswers };
-								const finalAnswers = this._arrangeAnswersForImportObjects(
-									combinedAnswers
-								);
-								resolve(finalAnswers);
-							});
-						})
-						.catch(error => {
-							reject(
-								TranslationService.getMessage(
-									ERRORS.CALLING_LIST_OBJECTS,
-									NodeUtils.lineBreak,
-									error
-								)
-							);
-						});
-				})
-				.catch(error =>
-					reject(
-						TranslationService.getMessage(
-							PROMPTING_INTERACTIVE_QUESTIONS_FAILED,
-							NodeUtils.lineBreak,
-							error
-						)
-					)
-				);
-		});
+	async _getCommandQuestions(prompt) {
+		try {
+			const listObjectQuestions = this._generateListObjectQuestions();
+			const listObjectAnswers = await prompt(listObjectQuestions);
+
+			const paramsForListObjects = this._arrangeAnswersForListObjects(listObjectAnswers);
+			const executionContextForListObjects = new SDKExecutionContext({
+				command: this._listObjectsMetadata.name,
+				params: paramsForListObjects,
+				includeProjectDefaultAuthId: true,
+			});
+
+			let operationResult;
+			try {
+				operationResult = await executeWithSpinner({
+					action: this._sdkExecutor.execute(executionContextForListObjects),
+					message: TranslationService.getMessage(MESSAGES.LOADING_OBJECTS),
+				});
+			} catch (error) {
+				return Promise.reject(TranslationService.getMessage(ERRORS.CALLING_LIST_OBJECTS, NodeUtils.lineBreak, error));
```

Comment:
minor one... you can extract some variable for not having everything in one line f(also the same below)

Reply:
Outdated as now the catch is throwing an exception and  the line is shorter